packages/typespec-vscode/src/code-action-provider.ts

```diff
@@ -46,10 +50,66 @@ export class TypeSpecCodeActionProvider implements vscode.CodeActionProvider {
           ),
         );
       }
-    });
+
+      // When the corresponding node dependency package is not installed,
+      // consider generating a quick fix to install via npm command
+      if (
+        diagnostic.source === "TypeSpec" &&
+        diagnostic.code &&
+        diagnostic.code === "import-not-found" &&
+        "data" in diagnostic &&
+        diagnostic.data &&
+        typeof diagnostic.data === "object" &&
+        "file" in diagnostic.data
+      ) {
+        // The message content is `Couldn't resolve import "@typespec/http"`.
+        // The compiler's diagnostics do not provide a specific attribute to obtain the package name,
+        // so a regular expression is used to extract the package name within the double quotes,
+        // if no match is reached, the original string is returned
+        const targetPackage = diagnostic.message.replace(/.*"([^"]+)".*/, "$1");
+        logger.debug(`The target package name is '${targetPackage}'.`);
+
+        const packageFile = diagnostic.data.file as string;
+        const packageFileUri = vscode.Uri.parse(packageFile);
+        const { packageJsonFolder, packageJson } = await searchAndLoadPackageJson(
+          getDirectoryPath(packageFileUri.fsPath),
+        );
+
+        if (
+          packageJsonFolder === undefined ||
+          (!targetPackage.startsWith("./") &&
+            !targetPackage.startsWith("../") &&
+            !isPathAbsolute(targetPackage) &&
+            packageJson &&
+            ((packageJson.peerDependencies && targetPackage in packageJson.peerDependencies) ||
+              (packageJson.dependencies && targetPackage in packageJson.dependencies) ||
+              (packageJson.devDependencies && targetPackage in packageJson.devDependencies)))
```

Comment:
is this just going to install the package if its already in the package.json dependencies?